pretty_midi/pretty_midi.py

```diff
@@ -402,6 +403,161 @@ def estimate_tempo(self):
         """
         return self.estimate_tempi()[0][0]
 
+    def get_beats_using_metadata(self):
+        """Uses Time Signature and Tempo metadata to estimate beat times
+
+        Returns
+        -------
+        beat_times : np.ndarray
+            List of beat times in seconds
+        """
+
+        # threshold in seconds for merging beat locations
+        epsilon = 0.05
+
+        # get qpms (quarter per minute) tempi and their change time
+        tempo_change_times, qpms = self.get_tempo_changes()
+
+        # container for storing beat times
+        beat_times = []
+
+        # store time signature objects
+        ts_idx = 1
+        if len(self.time_signature_changes) > 1:
+            cur_ts = self.time_signature_changes[0]
+            nxt_ts = self.time_signature_changes[1]
+        else:
+            cur_ts = self.time_signature_changes[0]
+            nxt_ts = None
+
+        # go over tempo change times and time signatures assuming that the first
+        # tempo change time is the start time of the first beat
+        i = 1
```

Comment:
For consistency/readability it would be cool if `i` was renamed `tempo_idx` or something.  I did this wrong in `get_beats` and considered changing it but haven't yet... now would not be a bad time!